docs/guide/check.qmd

```diff
@@ -0,0 +1,329 @@
+    "version": "1.0.0",
+    "licenses": [{"name": "odc-pddl"}, {"name": "mit"}],
+    "resources": [
+        {
+            "name": "woolly-dormice-2015",
+            "title": "Body fat percentage in the hibernating woolly dormouse",
+            "path": "resources/woolly-dormice-2015/data.parquet",
+        }
+    ],
+}
+
+config = cdp.Config(rules=[license_rule])
+issues = cdp.check(descriptor=package_descriptor, config=config)
+print(issues)
+```
+
+We can see that the custom rule was applied: `check()` returned one
+issue flagging the first license attached to the Data Package.
+
+### Strict mode
+
+The Data Package standard expresses both required and recommended rules.
+By default, `check()` checks only required rules. Recommended rules can
+be turned on by setting the `strict` argument to `True`. The example
+below violates the recommendation that the package `name` should contain
+no special characters.
+
+```{python}
+package_descriptor = {
+    "name": "Woolly Dormice (Toros Dağları)",
+    "title": "Hibernation Physiology of the Woolly Dormouse: A Scoping Review.",
+    "description": dedent("""
+        This scoping review explores the hibernation physiology of the
+        woolly dormouse, drawing on data collected over a 10-year period
+        along the Taurus Mountain range in Turkey.
+        """),
+    "id": "123-abc-123",
+    "created": "2014-05-14T05:00:01+00:00",
+    "version": "1.0.0",
+    "licenses": [{"name": "odc-pddl"}],
+    "resources": [
+        {
+            "name": "woolly-dormice-2015",
+            "title": "Body fat percentage in the hibernating woolly dormouse",
+            "path": "resources/woolly-dormice-2015/data.parquet",
+        }
+    ],
+}
+
+issues = cdp.check(descriptor=package_descriptor, strict=True)
+print(issues)
+```
+
+## Understanding `Issue`s
+
+### The `Issue` class
+
+One `Issue` corresponds to one failed check on one field in the
+descriptor. An `Issue` has three attributes:
+
+-   `type`: The type of the check that failed and caused the issue. This
+    can be a type that exists in the Data Package standard (e.g.,
+    `required` or `pattern`) or a custom type.
+-   `location`: The location of the field that failed the check
+    expressed in [JSON path](https://en.wikipedia.org/wiki/JSONPath)
+    notation. For example, `$.resources[2].name`.
+-   `message`: An explanation of what went wrong.
+
+### The `explain()` function
+
+The `explain()` function provides a more verbose and user-friendly
+description of a list of `Issue`s. To use it, pass it the issues output
+by `check()`:
+
+```{python}
+package_descriptor = {
+    "name": 123,
+    "title": "Hibernation Physiology of the Woolly Dormouse: A Scoping Review.",
+    "id": "123-abc-123",
+    "created": "2014-05-14T05:00:01+00:00",
+    "version": "1.0.0",
+    "licenses": [{"name": "odc-pddl"}],
+}
+
+issues = cdp.check(descriptor=package_descriptor)
+print(cdp.explain(issues))
+```
```

Comment:
````suggestion
```

When using `check(error=True)`, the `explain()` function will be used by default as
the error messages.
````